enterprise_access/apps/content_assignments/management/commands/send_assigment_deadline_passed_emails.py

```diff
@@ -0,0 +1,87 @@
+"""
+Management command to send email to learners for whom assignment has passed deadline.
+"""
+
+import logging
+from time import sleep
+
+from django.utils import timezone
+from django.core.management.base import BaseCommand
+
+from django.db.models import Q
+from enterprise_access.apps.content_assignments.models import LearnerContentAssignment
+from enterprise_access.apps.content_assignments.tasks import send_assignment_deadline_passed_email
+from enterprise_access.apps.content_assignments.constants import LearnerContentAssignmentStateChoices
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Send email to learners for whom assignment has passed deadline
+    """
+    help = (
+        'Spin off celery tasks to send learners an email for whom assignment has passed deadline'
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--batch-size',
+            action='store',
+            dest='batch_size',
+            default=25,
+            help='How many tasks to kick start before sleeping.',
+            type=int,
+        )
+        parser.add_argument(
+            '--sleep-duration',
+            action='store',
+            dest='sleep_duration',
+            default=5,
+            help='How long to sleep between batches.',
+            type=int,
+        )
+
+    def _we_should_sleep(self, task_number, batch_size):
+        return task_number % batch_size == 0
+
+    def get_deadlinse_passed_assignments(self):
+
+        # ASSUMPTION: `ALLOCATED` means the learner hasn't used the assignment
+        allocated_assignments = Q(status=LearnerContentAssignmentStateChoices.ALLOCATED)
+
+        # Check 90 days have passed after assignment
+        ninety_days_old = timezone.now() - timezone.timedelta(days=90)
+        allocated_and_ninety_days_have_passed = Q(
+            status=LearnerContentAssignmentStateChoices.ALLOCATED,
+            created__lt=ninety_days_old
+        )
+
+        # combine the filters
+        # can this introduce duplicates?
+        queryset = LearnerContentAssignment.objects.filter(
+            allocated_assignments | allocated_and_ninety_days_have_passed
+        )
+
+        # TODO: Call discovery to check course enrollment end date has passed
+        # get enrollment end date for all assignments in the `queryset` and then filter the `queryset`
+
+        # TODO: Check contract expiration date has passed
+        # Not sure from where to fetch this?
+
+        return queryset.values_list(
+            'uuid',
+            flat=True,
+        )
+
+    def handle(self, *args, **options):
```

Comment:
1. I would iterate over `AssignmentConfiguration` records at the top-level, and then iterate over the allocated assignments for that configuration.  An `AssignmentConfiguration` relates to a `SubsidyAccessPolicy`, which means we can iterate through one budget at a time - probably useful for thinking about logging the progress of this command.
2. We need to make sure that the assignment record has its state updated to `CANCELLED` when it is expired.
3. You can use `get_content_metadata_for_assignments()` function to fetch metadata about an assignment's content.  The place where that lives is being refactored a little right now: https://github.com/openedx/enterprise-access/blob/b17e09e485571ea8efa6f17bd3dc3bd564eac7da/enterprise_access/apps/content_assignments/content_metadata_api.py but the function signature and functionality should remain unchanged.
4. "Contract expiration" really just means that the related subsidy record has expired, which you can check via https://github.com/openedx/enterprise-access/blob/ddd745088f58e97c51175f40836b105d8f66757f/enterprise_access/apps/subsidy_access_policy/models.py#L197
Here's some pseudocode for how'd I would generally go about executing this.
```python
for each active assignment_configuration:
  assignments = assignment_configuration.assignments.filter(state=ALLOCATED)
  for each assignment in assignments:
    if current_date() > assignment.created + 90 days:
      assignment.expire() # this method doesn't exist, but we might want it.  make sure to update the state to cancelled
      send_assignment_deadline_passed_email.delay(assignment.uuid)
    metadata_dict = get_your_metadata_lookup_by_content_key()
    if metadata_dict[assignment.content_key].get('enroll_by_date') < current_date():
        # expire and invoke email task
    if assignment_configuration.policy.subsidy_expiration_date() < current_date():
        # expire and invoke email task
```

One more thing that's in the weeds: make sure to not give `get_content_metadata_for_assignments()` more than 100 assignments as input.  Under the covers, it relies on our enterprise-catalog metadata fetching endpoint, which will return up to 100 metadata results in a single request.